### ISSUE TITLE
Add audio recording with sync option to 2d viewer

### ISSUE THREAD

**Issue by ReneRebsdorf** [akiver/cs-demo-manager#952]
### What problem is this solving

Being able to listen to communications from a discord or teamspeak server would provide great insight and context to the analyzed video

### Proposed solution

- Add option for matches to allow referencing an mp3 (or other audio format)
- Provide option to sync the audio to the 2d video manually (recordings are unlikely to start at the same time as match recording)
- Based on known factors for how long it takes from a round to start after "round end", it should be possible to align each round with the correct time for the audio recording

### Describe alternatives you've considered

N/A

**Comment by ReneRebsdorf** [comment]
@akiver what are your thoughts on this? Is this something that would potentially be merged into cs-demo-manager or is it out of scope?

**Comment by ReneRebsdorf** [comment]
I see, solutions such as [Skybox Edge](https://edge.skybox.gg/)

It would be great to have an open source alternative, and based on my research your cs-demo-manager is what comes closest.
I was in touch with Scope.gg and Leetify, but both have accuracy issues with their videos (e.g. during video playback a second is not accurate to a real life second), which of course is a big blocker for adding external audio. My tests for your solution proved to be accurate.

**Comment by ReneRebsdorf** [comment]
@akiver something I didn't mention previously is that yeah: the synching is annoying to do, but it is worth it for those of us who use this as part of team analysis, and once you get into a habit it isn't too bad. Just providing a number in 10th of a second to delay the video to delay (allow negatives and positives), then we would just take note of how delayed my audio is by saying "Go" at the start of the 1st round (after freezetime). Other more fancy methods probably exists too.

Wrt. to changing rounds, it also has to account for pauses such as timeouts, and the 2d replayer would need to be able to auto-start each round.

**Comment by akiver** [comment]
Just a heads-up, I will tackle [this feature request](https://github.com/akiver/cs-demo-manager/discussions/1099) before starting to work on this one.
It will help me since I will be able to generate the audio file for a full match length file.
I have a draft that works for CSGO demos, and have to make it work for CS2 demos.

**Comment by ReneRebsdorf** [comment]
> Just a heads-up, I will tackle [this feature request](https://github.com/akiver/cs-demo-manager/discussions/1099) before starting to work on this one.
> It will help me since I will be able to generate the audio file for a full match length file.
> I have a draft that works for CSGO demos, and have to make it work for CS2 demos.

That's awesome! Thank you for the update

**Comment by akiver** [comment]
It will be available in the next version, I added a doc about this feature https://cs-demo-manager.com/docs/guides/2d-viewer

**Comment by ReneRebsdorf** [comment]
That is awesome! I can't wait to try it out

Amazing work @akiver
